File: storage/data_original/at_coder/arc051/A/4837000.py
x,y,r = map(int,input().split())
a,b,c,d = map(int,input().split())

if a <= x-r and x+r <= c and b <= y-r and y+r <= d:
    print("NO")
else:
    print("YES")

if (c-x)**2+(d-y)**2 <= r**2 and (c-x)**2+(b-y)**2 <= r**2 and (a-x)**2+(d-y)**2 <= r**2 and (a-x)**2+(b-y)**2 <= r**2:
    print("NO")
else:
    print("YES")

File: storage/data_transpiled/at_coder/arc051/A/4837000.java
public static void solve ( ) {
  int x = Integer . parseInt ( input . nextLine ( ) ) ;
  int y = Integer . parseInt ( input . nextLine ( ) ) ;
  int r = Integer . parseInt ( input . nextLine ( ) ) ;
  int a = Integer . parseInt ( input . nextLine ( ) ) ;
  int b = Integer . parseInt ( input . nextLine ( ) ) ;
  int c = Integer . parseInt ( input . nextLine ( ) ) ;
  int d = Integer . parseInt ( input . nextLine ( ) ) ;
  if ( a <= x - r && x + r <= c && b <= y - r && y + r <= d ) {
    System . out . println ( "NO" ) ;
  }
  else {
    System . out . println ( "YES" ) ;
  }
  if ( ( c - x ) * ( c - x ) + ( d - y ) * ( c - x ) * ( c - x ) + ( b - y ) * ( b - y ) * ( a - x ) + ( d - y ) * ( d - y ) * ( a - x ) + ( b - y ) * ( b - y ) * ( a - x ) ) {
    System . out . println ( "NO" ) ;
  }
  else {
    System . out . println ( "YES" ) ;
  }
}
